content/zh-cn/designs/telemetry/_index.md

```diff
@@ -0,0 +1,168 @@
+
+当实现上述设计时，请考虑如下问题：
+
+#### 在IoT解决方案中，遥测消息所需的 *感测到洞察*(sense-to-insight) 或 *感测到行动*(sense-to-action) 处理延时是多少？
+
+对于IoT解决方案中需要**微秒**或**毫秒**级别处理延时的，相应的处理应该在设备本身或是连接到设备的设备[网关]({{<ref "/designs/gateway" >}})上进行。
+
+对于IoT解决方案中需要**秒**,**分钟**甚至是**小时**级别处理延时的，相应的处理默认应该在云端进行。通常来说，对于需要“秒”到“分钟”级别处理的消息，应该由直接连接到协议端点的组件来进行处理。在消息到达并满足一定的条件时，相应的组件会被触发并对消息进行处理。
+
+对于从“分钟”到“小时”延时级别的遥测数据，应该以异步的方式进行处理。当消息到达并满足期望的条件时，相应的事件会被放在一个处理队列中并且有对应的组件进行必要的处理。一旦处理完成，该组件通常会发送一条消息到"处理完成"[消息主题]({{< ref "/glossary/vocabulary#message-topic" >}}).
+
+#### 是否有经验教训可以使IoT解决方案中的遥测数据更易于处理？
+
+**解决方案唯一的设备ID** - 每个设备在解决方案中都应该有*解决方案唯一*的ID。虽然这个ID并不一定真的要全球唯一，但每个设备需要在IoT解决方案中有永远唯一的ID。通过支持解决方案唯一的设备ID，IoT解决方案可以更好的处理和路由感测的数据，供解决方案内的组件所使用。
+
+**尽早打时间戳** - 越早让感测数据获得在IoT解决方案中离散的时间戳，就越容易对感测数据进行细致入微的处理和分析
+
+**闭合窗口计算** - 追踪设备的 `last_reported`(最后报告)时间戳可以确定是否/何时聚合窗口可以被视为*闭合*。然后在IoT解决方案中可以轻松且可信的缓存任何闭合窗口计算结果。这些缓存的计算结果经常可以极大的提高IoT解决方案中*感测到洞察(sense-to-insight)* 或 *感测到行动(sense-to-action)* 的处理性能。
+
+#### 大消息应该如何处理
+在这个设计中，大消息被定义为任何大于传输协议原生能支持的消息。对于大消息，需要回答一个额外的问题：是否可以使用辅助协议？
+
+如果**是**，则推荐使用HTTP协议
+
+如果**不是**,则大消息需要被切片，每个分片需要有一个独立的切片ID，同时每个切片都需要足够小，以便以能够使用传输协议进行发送
+
+#### 当使用辅助协议时，应该如何发送大消息
+如果大消息必须**尽可能快的传递**，则可以直接将大消息上传至一个全球可用，且具有强数据持久性的对象存储服务
+
+如果大消息**可以被批量发送**, 那每个消息应该做为批量消息的一部分保存下来，直到批量消息可以被发送出去。由于设备上的存储资源通常是非常有限，对消息的批量处理应该考虑与设备做为[网关]({{<ref"/designs/gateway" >}})时同样的[算法权衡]({{< ref "/designs/gateway#how-should-data-be-processed-when-the-internet-is-unavailable" >}})
+
+#### 什么是设备的采样和报告频率？
+
+**采样频率** 是从连接的[传感器]({{< ref "/glossary/vocabulary#sensor" >}})上获取或采样感测数据的频率。
+
+**报告频率** 是存储在设备上的采样数据发送到IoT解决方案的频率
+  
+设备上的代码会获取感测数据并将其排队以进行传递，或立即传递感测数据。这两种不同的行为通常被讨论为*采样频率和报告频率之间的差异*。当采样和报告频率相等且对齐时，所有感测数据将被立即传递。当两个频率不同时，必须考虑为入队数据选择正确的日志算法。
+
+这两个频率的预期数值对于确定IoT解决方案的规模和成本是至关重要的。
+
+#### 是否需要维持入站消息的顺序
+首先，解决方案只有在绝对必要时，才应该依赖顺序  
+
+如果**不要求**顺序，那解决方案可以在消息到达后立即进行处理
+
+如果**要求**顺序，那么需要回答以下问题："在多长一段时间范围内，解决方案的组件需要有序的消息？"
+
+如果答案是"在一个主题上小于1秒钟"，那解决方案可以从主题`foo`上将消息收集到一个缓冲区，然后每1秒钟缓冲区的消息被排序并且发送到另一个主题 `foo/ordered`。
+
+如果上述答案是“大于1秒”，那IoT解决方案应该将每个记录都写入一个[有序存储]({{< ref "/glossary/vocabulary#ordered-store" >}})中。解决方案中任何**要求**消息总是有序的组件，可以从有序存储中读取并获得更新。
+
+#### IoT解决方案中遥测的成本驱动因素有哪些？
+
+通常，物联网解决方案中最常见的成本驱动因素是设备数量、设备采样和报告频率、必要的*感知到洞察力*或*感知到行动*的遥测处理延迟、[设备数据密度] ({{<ref"/glossary/vocabulary#device-data-density">}})以及[遥测存档]({{<ref "/designs/telemetry_archiving" >}})的保留时间。
+
+#### 每个设备是否与其他设备在报告频率上“主动不对齐”？
+当IoT解决方案中的所有设备都配置相同的报告频率时，则会出现一个具有重大影响的常见错误。为了避免隐藏在这个简单行为中的[相长干涉](http://en.wikipedia.org/w/index.php?title=Constructive_interference)，设备应该只在它唤醒并且在一段随机持续时间过后才开始进行报告。这种启动时间的随机性可以产生一个更为平滑的进入解决方案的感测数据流，防止因为设备在不可避免的区域网络中断或其他对解决方案有影响的中断后恢复通信时产生的相长干涉。
+
+#### 当设备无法连接到其默认的IoT解决方案端点时应该做些什么？
+
+**可预期的中断时长** - 当设备在可预期的时长内无法连接到默认的IoT解决方案端点，该设备应该按照配置进行*设备消息排队*。这与确定设备采样和报告频率的差异时的答案可能是一样的。此外，任何能够执行设备消息排队的设备都应该考虑与做为设备[网关]({{<ref"/designs/gateway">}})的设备相同的算法权衡。当本地存储不足以在预期持续时间内存储所有消息并且将影响所感测的数据时，则需要做出权衡。通常需要考虑的算法类别包括**[先进先出(FIFO)](https://en.wikipedia.org/wiki/FIFO_(computing_and_electronics))**、**剔除(Culling)**和**聚合(Aggregate)**
```

Comment:
Please ensure that these Wikipedia URLs are pointing to the Simplified Chinese version on Wikipedia (if that version exists).